mart/attack/adversary.py

```diff
@@ -179,15 +181,20 @@ def forward(self, *, batch: torch.Tensor | list | dict, model: Callable):
         self.attacker.fit_loop.max_epochs += 1
         self.attacker.fit(self, train_dataloaders=cycle([batch_and_model]))
 
-        # Get the input_adv for enforcer checking.
+    def forward(self, *, batch=None, input=None, target=None, enforce=True):
+        """Compose adversarial examples and revert to the original input format."""
+        if batch is not None:
+            input, target = self.batch_c15n(batch)
+
+        # Get the canonicalized input_adv for enforcer checking.
         perturbation = self.perturber(input=input, target=target)
         input_adv = self.composer(perturbation, input=input, target=target)
-        self.enforcer(input_adv, input=input, target=target)
 
-        # TODO: We assume batch is a tuple, but it may be different for models outside MART. Add a reverse transform?
-        # Revert to the original format of batch.
-        batch_adv = (input_adv, target)
+        if enforce:
```

Comment:
Why not always enforce?

Reply:
Fixed in [fd0e847](https://github.com/IntelLabs/MART/pull/215/commits/fd0e8472c08e9e9930c8240adef4f5b1e92e5812)

Simplicity over (negligible) efficiency.